scripts/get_collection_statics.py

```diff
@@ -0,0 +1,242 @@
+    grouped_df.drop(['first_letter'], axis=1, inplace=True)
+    grouped_df['mapped_callnoletters'] = grouped_df['mapped_callnoletters'].apply(lambda x: get_category_name(x))
+
+    # Create a new column to store the mapped values
+    df.drop(['Percent'], axis=1, inplace=True)
+    df.drop(['first_letter'], axis=1, inplace=True)
+    df.sort_values(by='value', ascending=True, inplace=True)
+    df['value'] = df['value'].str.upper()
+
+    return df, grouped_df
+
+def create_dataframe_from_facets(facet_fields):
+    """
+    Create a DataFrame from Solr facet fields output.
+    :param facet_fields: Dictionary containing facet fields and their counts.
+    :return: DataFrame with facet fields and counts.
+    """
+    facet_data = {}
+
+    for field, values in facet_fields.get('facet_fields').items():
+        data = []
+        for i in range(0, len(values), 2):
+            data.append({
+                'value': values[i],
+                'Count': values[i + 1]
+            })
+        facet_data[field] = data
+
+    return facet_data
+
+
+def add_percentage_column(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a column with the percentage of each row in the DataFrame.
+    :param df: DataFrame containing facet fields and counts.
+    :return: DataFrame with an additional percentage column.
+    """
+    total_count = df['Count'].sum()
+    df['Percent'] = ((df['Count'] / total_count) * 100).round(2)
+    return df
+
+
+def sort_dataframe_by_percentage(df):
+    """
+    Sort the DataFrame by the percentage column in descending order.
+    :param df: DataFrame containing facet fields, counts, and percentages.
+    :return: Sorted DataFrame.
+    """
+    df_sorted = df.sort_values(by='Percent', ascending=False)
+    return df_sorted
+
+
+def create_copyrights_status(df: pd.DataFrame) -> pd.DataFrame:
+
+    """Transform the right status into a more readable format
+    :param df: DataFrame containing the right status and counts.
+    :return: DataFrame with the right status in a more readable format.
+    """
+
+    # Ignoring these categories 4,6,16, 8, 26, 27
+    rights_status = {
+        "Public Domain worldwide": ['1', '17'],
+        "In Copyright": ['2', '3', '5'],
+        "Public Domain in the US": ['9'],
+        "in copyright only in the US": ['19'],
+        "creative commons or open access": ['7', '10', '11', '12', '13', '14', '15', '18', '20', '21', '22', '23', '24',
+                                            '25'],
+    }
+
+    df_new = pd.DataFrame()
+    for key, values in rights_status.items():
+        filtered = df[df['value'].isin(values)]
+        filtered['category'] = filtered["value"].apply(lambda x: key)
+        df_new = pd.concat([df_new, filtered], ignore_index=True)
+
+    df_new = df_new.groupby('category')['Count'].sum().reset_index()
+    return df_new
+
+
+def create_excel_file(solr_facets_output: dict, file_name: str, map_file_path: str = None):
```

Comment:
For future reusability for other purposes it may be preferable just to create tsv file(s) rather than Excel, but the Excel sheet is certainly easier to upload to Google Drive 🤷

Reply:
Using the Google Sheets API to write data directly to our Google Drive would be a better solution. I would choose this approach if needed for other use cases.

Reply:
That's also an option -- although it might be easier to write the excel file and then use the Drive API to upload the file. I think we need to see what the use cases are. With the old web site, we had some tsv files with statistics and then some javascript to render those.

Reply:
I will add a TODO to this line. In the future we could apply this change